source/predicates/predicate.ts

```diff
@@ -4,6 +4,14 @@ import {not} from '../operators/not';
 import {BasePredicate, testSymbol} from './base-predicate';
 import {Main} from '..';
 
+/**
+Function executed when the provided validation fails.
+The first argument provided to the function is the provided `value` for the property,
+the second argument is the optional `label` for the property.
+The returned value will be the error message.
+ */
+export type ValidatorMessage<T> = (value: T, label?: string) => string;
```

Comment:
It's not a validator message though, it creates the validator message. I think the name here could be improved.

Reply:
Changed it to `ValidatorMessageBuilder`, couldn't really think on anything better. Any suggestions perhaps ?